archiver.go

```diff
@@ -89,6 +90,19 @@ func (a *Archiver) do(log log15.Logger, j *Job) (err error) {
 		"last-fetch", r.FetchedAt,
 		"references", len(r.References))
 
+	if r.Status == model.Fetching {
```

Comment:
If we move this to a method it will become:

```go
if err := a.foo(); err != nil {
  return err
}
```
or 

```go
if a.foo() {
  return ErrAlreadyFetching.New(blabla)
}
```

Which is exactly as long as it is now (excluding the log, that I would not move to the method). Not sure what benefit would we obtain from moving this to a method